lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -321,6 +322,22 @@ def add_payment(data, card)
         end
       end
 
+      def add_external_mpi_fields(data, options)
+        return unless options[:three_d_secure]
+
+        data[:txid] = options[:three_d_secure][:xid] if options[:three_d_secure][:xid]
+        data[:cavv] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
+        data[:eci] = options[:three_d_secure][:eci] if options[:three_d_secure][:eci]
+
+        data[:threeDSServerTransID] = options[:three_d_secure][:xid] if options[:three_d_secure][:xid] 
+        data[:dsTransID] = options[:three_d_secure][:ds_transaction_id] if options[:three_d_secure][:ds_transaction_id]
+        data[:protocolVersion] = options[:three_d_secure][:version] if options[:three_d_secure][:version]
+
+        data[:authenticacionMethod] = options[:three_d_secure][:eci] if options[:three_d_secure][:authentication_method]
+        data[:authenticacionType] = options[:three_d_secure][:eci] if options[:three_d_secure][:authentication_type]
+        data[:authenticationFlow] = options[:three_d_secure][:eci] if options[:three_d_secure][:authentication_flow]
```

Comment:
Since these are optional fields for Redsys that do not map to any of our standardized 3DS fields, should we consider allowing the customer to pass these in as standard GSFs that are provided outside of the `options[:three_d_secure]` hash?